Client/Frontend/Widgets/PhotonActionSheet/PhotonActionSheet.swift

```diff
@@ -4,81 +4,58 @@
-        tapRecognizer.numberOfTapsRequired = 1
-        tapRecognizer.cancelsTouchesInView = false
-        tapRecognizer.delegate = self
-        return tapRecognizer
-    }()
-
-    lazy var closeButton: UIButton = {
-        let button = UIButton()
+    // MARK: - Variables
+    private var tableView = UITableView(frame: .zero, style: .grouped)
+    private var viewModel: PhotonActionSheetViewModel!
+    private var constraints = [NSLayoutConstraint]()
+
+    private lazy var closeButton: UIButton = .build { button in
         button.setTitle(.CloseButtonTitle, for: .normal)
         button.setTitleColor(UIConstants.SystemBlueColor, for: .normal)
         button.layer.cornerRadius = PhotonActionSheetUX.CornerRadius
         button.titleLabel?.font = DynamicFontHelper.defaultHelper.DeviceFontExtraLargeBold
-        button.addTarget(self, action: #selector(dismiss), for: .touchUpInside)
+        button.addTarget(self, action: #selector(self.dismiss), for: .touchUpInside)
         button.accessibilityIdentifier = "PhotonMenu.close"
-        return button
-    }()
+    }
 
     var photonTransitionDelegate: UIViewControllerTransitioningDelegate? {
         didSet {
-            self.transitioningDelegate = photonTransitionDelegate
+            transitioningDelegate = photonTransitionDelegate
         }
     }
 
-    init(site: Site, actions: [PhotonActionSheetItem], closeButtonTitle: String = .CloseButtonTitle) {
-        self.site = site
-        self.actions = [actions]
-        self.style = .centered
-        super.init(nibName: nil, bundle: nil)
-        self.closeButton.setTitle(closeButtonTitle, for: .normal)
-    }
+    // MARK: - Init
 
-    init(title: String? = nil, actions: [[PhotonActionSheetItem]], closeButtonTitle: String = .CloseButtonTitle, style presentationStyle: UIModalPresentationStyle? = nil) {
-        self.actions = actions
-        if let presentationStyle = presentationStyle {
-            self.style = presentationStyle == .popover ? .popover : .bottom
-        } else {
-            self.style = .centered
-        }
+    init(viewModel: PhotonActionSheetViewModel) {
+        self.viewModel = viewModel
         super.init(nibName: nil, bundle: nil)
-        self.title = title
-        self.closeButton.setTitle(closeButtonTitle, for: .normal)
 
-        self.tableView.estimatedRowHeight = PhotonActionSheetUX.RowHeight
+        title = viewModel.title
+        modalPresentationStyle = viewModel.modalStyle
+        closeButton.setTitle(viewModel.closeButtonTitle, for: .normal)
+        tableView.estimatedRowHeight = PhotonActionSheetUX.RowHeight
     }
 
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
+    deinit {
+        tableView.dataSource = nil
+        tableView.delegate = nil
+        NotificationCenter.default.removeObserver(self)
```

Comment:
NotifiableProtocol!

Reply:
Notifiable handles setup but not removeObserver, so I think I have to keep that